src/app/Analytics/SRE_Analytics/RecommendationComponents/ChatDataPreparation.tsx

```diff
@@ -0,0 +1,90 @@
+// export function filterLastNDayData(data, givenDay, numDays) {
+//   const oneDayInMillis = 24 * 60 * 60 * 1000; // Number of milliseconds in a day
+//   const givenDayMillis = Date.parse(givenDay);
+//   const filteredData = {};
+
+//   for (const key in data) {
+//     if (data.hasOwnProperty(key)) {
+//       const timestampMillis = Date.parse(key);
+//       if (givenDayMillis - timestampMillis <= oneDayInMillis * numDays && givenDayMillis - timestampMillis >= 0) {
+//         filteredData[key] = data[key];
+//       }
+//     }
+//   }
+
+//   return filteredData;
+// }
```

Comment:
Why is this commented?

Reply:
This is the function which gets the last N days data including the selected timestamp, currently we use another function with the same name as this one but that function gives N days data excluding selected timestamp.

 I have commented it because I will be using this next to fix the v-shape wrong display issue discussed [on slack](https://kruizeworkspace.slack.com/archives/C0380EU8J2H/p1697096736198609)